libcontainer/cgroups/fs2/psi.go

```diff
@@ -0,0 +1,87 @@
+package fs2
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"golang.org/x/sys/unix"
+
+	"github.com/opencontainers/runc/libcontainer/cgroups"
+)
+
+func statPSI(dirPath string, file string) (*cgroups.PSIStats, error) {
+	f, err := cgroups.OpenFile(dirPath, file, os.O_RDONLY)
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			// Kernel < 4.20, or CONFIG_PSI is not set.
```

Comment:
Maybe also add the case when kernel >= 6.1 and PSI turned off per cgroup ?  The file also doesn't exist for the cgroup as well in this case

https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=bd9a3dba185ce6701b41f0341470d3f53bbbbaed

Reply:
Added, thanks!

```diff
diff --git a/libcontainer/cgroups/fs2/psi.go b/libcontainer/cgroups/fs2/psi.go
index 0d5f1d7a..09f34888 100644
--- a/libcontainer/cgroups/fs2/psi.go
+++ b/libcontainer/cgroups/fs2/psi.go
@@ -17,7 +17,9 @@ func statPSI(dirPath string, file string) (*cgroups.PSIStats, error) {
        f, err := cgroups.OpenFile(dirPath, file, os.O_RDONLY)
        if err != nil {
                if errors.Is(err, os.ErrNotExist) {
-                       // Kernel < 4.20, or CONFIG_PSI is not set.
+                       // Kernel < 4.20, or CONFIG_PSI is not set,
+                       // or PSI stats are turned off for the cgroup
+                       // ("echo 0 > cgroup.pressure", kernel >= 6.1).
                        return nil, nil
                }
                return nil, err

```